src/GAP/gap_to_oscar.jl

```diff
@@ -240,7 +240,7 @@ end
 ## GAP straight line program
 function straight_line_program(slp::GapObj)
   @req GAP.Globals.IsStraightLineProgram(slp) "slp must be a straight line program in GAP"
-  lines = GAP.gap_to_julia(GAP.Globals.LinesOfStraightLineProgram(slp); recursive = true)
+  lines = Vector{Any}(GAP.Globals.LinesOfStraightLineProgram(slp))
```

Comment:
Does this work? It might avoid a conversion when constructing the `SLP.GAPSLProgram`
```suggestion
  lines = Vector{GAPStraightLine}(GAP.Globals.LinesOfStraightLineProgram(slp))
```

Reply:
nope, I tried that. But the conversion code is not capable of deciding which of the union elem types to convert each element of the vector to

Reply:
OK makes sense.